x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -451,7 +452,7 @@ else if (plan instanceof OrderBy o) {
 
         private List<NamedExpression> filterUnsupportedProjections(List<NamedExpression> projections) {
             return projections.stream()
-                .filter(p -> p.resolved() == false || isTypeSupportedInVersion(p.dataType(), configuration.version()))
+                .filter(p -> p.resolved() == false || isTypeSupportedInVersion(p.dataType(), Version.fromId(configuration.version().id)))
```

Comment:
Why not use directly `configuration.version()` instead of `Version.fromId(configuration.version().id)` ?